clippy_lints/src/mut_key.rs

```diff
@@ -132,8 +133,14 @@ impl<'tcx> MutableKeyType<'tcx> {
             )
         {
             let subst_ty = args.type_at(0);
-            if self.interior_mut.is_interior_mut_ty(cx, subst_ty) {
-                span_lint(cx, MUTABLE_KEY_TYPE, span, "mutable key type");
+            if let Some(chain) = self.interior_mut.interior_mut_ty_chain(cx, subst_ty) {
+                span_lint_and_then(cx, MUTABLE_KEY_TYPE, span, "mutable key type", |diag| {
+                    for ty in chain.iter().rev() {
+                        diag.note(with_forced_trimmed_paths!(format!(
+                            "... because it contains `{ty}`, which has interior mutability"
```

Comment:
Since type names can be quite long sometimes, wouldn't it be best to use `…` instead of `...` here? In monospace font the difference would be two characters wide:

```txt
... because it contains `UnsafeCell<usize>`, which has interior mutability
… because it contains `UnsafeCell<usize>`, which has interior mutability
```

Reply:
I went with the three dots because I think it's much more common in diagnostics (in fact, I think I've never actually seen a non-ASCII character like that in a diagnostic - rust-lang/rust#126597 adds unicode block drawing but it requires an extra `--error-format=human-unicode`).

[Some rustc examples](https://github.com/search?q=repo%3Arust-lang%2Frust+path%3Atests+%2F%5Enote%3A+%5C.%5C.%5C.%2F&type=code)
[Some cargo examples](https://github.com/search?q=repo%3Arust-lang%2Fcargo+%2F%5E+%2B%5C.%5C.%5C.%2F&type=code)

Reply:
There is a "…" in a diagnostic issued by `clippy_lints/src/methods/or_then_unwrap.rs`:

```
clippy_lints/src/methods/or_then_unwrap.rs
25:        title = "found `.or(Some(…)).unwrap()`";
32:        title = "found `.or(Ok(…)).unwrap()`";
```